replica/table.cc

```diff
@@ -1479,6 +1479,10 @@ table::seal_active_memtable(compaction_group& cg, flush_permit&& flush_permit) n
         for (;;) {
             std::exception_ptr ex;
             try {
+                // We need to be able to simulate a retry in s3 tests
+                if (utils::get_local_injector().enter("memtable_flush_should_retry")) {
+                    throw storage_io_error(EPERM, "EPERM fault injected to simulate retry");
+                }
```

Comment:
That's too "explicit" for test. Please, place the failure deep in the s3 client code (`client::do_retryable_request` looks to be good place), so that it throws `aws::exception` that corresponds to "bad key/secret" (`aws::aws_error_type::HTTP_UNAUTHORIZED`?) and then propagates up the stak in the EPERM/EACCESS form.

Reply:
I threw in `do_retryable_request` and `map_s3_client_exception(e);` code takes care of propagating it into `EACCES`.